### ISSUE TITLE
Redundant OpenReport requests when opening a thread

### ISSUE THREAD

**Issue by tgolen** [Expensify/App#51680]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.55-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @tgolen 
**Slack conversation:** Reported it in https://github.com/Expensify/App/issues/49162#issuecomment-2435969058

## Action Performed:
1. Have User 1 and User 2 in a chat together
2. User 1 adds a comment
3. User 1 opens the comment in a thread
4. User 1 adds a comment to the thread
5. User 2 opens the thread

## Expected Result:
User 2 should only have a single call to `OpenReport`

## Actual Result:
Use 2 has two calls to `OpenReport`

## Workaround:
Can the user still use Expensify without this being fixed? Yes, it is non-breaking.

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  

![image](https://github.com/user-attachments/assets/98a7c6b5-0638-4f89-9105-7c5f9d82448a)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by tgolen** [comment]
@twisterdotcom @situchan This was something I reported in https://github.com/Expensify/App/issues/49162#issuecomment-2435969058 and @nkdengineer already debugged it and wrote a quick proposal for it. I'm opening this issue so the problem can be fixed. We can skip looking at other proposals for this one.

@nkdengineer Can you please comment on this issue so I can assign it to you?

**Comment by nkdengineer** [comment]
@tgolen I'm here.

**Comment by nkdengineer** [comment]
Thanks, I will open the PR tomorrow morning.

**Comment by twisterdotcom** [comment]
Zim zam zoom. Here to watch the masters at work and pay out.

**Comment by c3024** [comment]
I can be the C+ here because I was the C+ auto-assigned for the linked issue. 😅

**Comment by situchan** [comment]
I think this is dupe of https://github.com/Expensify/App/issues/49948

**Comment by bernhardoj** [comment]
@tgolen Yes, this is a dupe of https://github.com/Expensify/App/issues/49948

**Comment by nkdengineer** [comment]
@c3024 The PR is [here](https://github.com/Expensify/App/pull/51710).

**Comment by tgolen** [comment]
OK, regarding the duplicate... this is kind of a tough one and I am on the fence.

The other issue was opened first, and @bernhardoj's proposal is almost the same as @nkdengineer's. However, that other issue moved a bit slower and @nkdengineer already has a PR written. 

I think it's probably best to close out the other issue and let @nkdengineer get paid for the PR that has already been created to fix this.

**Comment by bernhardoj** [comment]
@situchan and I already mentioned that this was a dupe before the PR was open. I've been explaining things on the other issue waiting for an approval. I think it would be unfair if the original issue is closed in favor of a dupe issue.

**Comment by twisterdotcom** [comment]
I'm open to just paying out for both contributors for their time, but we need to prioritise which one PR is pushed and reviewed.

**Comment by tgolen** [comment]
The PR from @nkdengineer is already merged, so I don't think anything needs prioritized here.

**Comment by twisterdotcom** [comment]
Nice okay. I think we should pay out:

- $250 for @nkdengineer as a contributor
- $250 for @c3024 as C+
- $125 for @bernhardoj for the work done in https://github.com/Expensify/App/issues/49948.

All after deploy + 7 days.

**Comment by c3024** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [x] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production
  - [ ] 2b. Reported on staging (deploy blocker)
  - [ ] 2c. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [x] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: https://github.com/Expensify/App/pull/15994/files#r1839487684

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: No discussion was started because this bug could not have been identified earlier with specific checklist items.

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


- None

### Test:


1. [User A] Open the chat with user B.
2. [User A] Send a message.
3. [User A] Click on "Reply in thread" and send some messages in this thread.
4. [User B] Open the DM with user A.
5. Open the browser Developer Tools > Network.
6. [User B] Open the thread created at step 3.
7. Verify that only one `OpenReport` request is made in the Network tab.

Do we agree 👍 or 👎

**Comment by tgolen** [comment]
Anyone should be able to check and see if they can reproduce it. @shubham1206agra can you give it a try first?

**Comment by twisterdotcom** [comment]
Okay, let's pay out for this OpenReport fix.

**Comment by twisterdotcom** [comment]
Just need to pay out the two offers.

**Comment by twisterdotcom** [comment]
@nkdengineer please accept the upwork offer. I will pay out when I'm going through my upwork inbox at some point this week.

**Comment by JmillsExpensify** [comment]
$125 approved for @bernhardoj
